### PR TITLE
Payments list

### PR DESCRIPTION
**Pull request by shalapatil**
## Notion card
https://www.notion.so/saeloun/User-should-be-able-to-add-a-payments-entry-against-an-invoice-b0777f8eb0b54b0e87a51393bebf32bc

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

**Review by gowsik-ragunath — APPROVED** [review]
LGTM! just one minor comment.

File: app/models/invoice.rb [hunk 1]
```diff
@@ -53,6 +53,7 @@ class Invoice < ApplicationRecord
   belongs_to :client
   has_many :invoice_line_items, dependent: :destroy
   has_one :company, through: :client
+  has_many :payments
```
Comment (gowsik-ragunath) [inline app/models/invoice.rb:56]: Can we add `dependent: :destroy` or `:nullify` to payments association?

Comment (akhilgkrishnan) [reply app/models/invoice.rb:56]: @shalapatil can you check Gowsik's comment

File: app/models/payment.rb [hunk 2]
```diff
@@ -44,6 +44,7 @@ class Payment < ApplicationRecord
   ]
 
   belongs_to :invoice
+  has_one :company, through: :invoice
```
Comment (akhilgkrishnan) [inline app/models/payment.rb:47]:
```suggestion
  delegate :company, to: :invoice, allow_nil: true
```

**Comment by supriya3105** [comment]
@shalapatil Please check Gowsik's comments and resolve the conflicts

**Review by gowsik-ragunath — APPROVED** [review]
LGTM!

**Comment by supriya3105** [comment]
@shalapatil Please resolve conflicts . @akhilgkrishnan Please review and mark the conversations as resolved where applicable.

**Review by akhilgkrishnan — APPROVED** [review]
LGTM

**Pull request merged**